start_time: 0
duration: 1000

## Is_flying checks
enable_arm_check: false # require armed before checking for is_flying
is_flying_threshold: 10.5 # if accel measurement exceed this magnitude, set is_flying
update_baro_velocity_threshold: 0.5

# Logging
log_prefix: "/tmp/roscopter/ekf/"
enable_log: true

## Measurement Processing
use_mocap: false
use_gnss: true
use_baro: false
use_range: false
use_compassing: true
use_zero_vel: false

R_zero_vel: [0.001, 0.001, 0.001, # vel
              0.001] # Altitude

## Initial Uncertainty
P0: [10.0, 10.0, 10.0, # pos
     0.01, 0.01, 0.01, # att
     0.001, 0.001, 0.001, # vel
     0.01, 0.01, 0.01, # accel bias
     0.001, 0.001, 0.001, # gyro bias
     9.0,           # baro bias
     100.0] # reference altitude

## Additive Process Noise
Qx: [0.000, 0.000, 0.000,       # pos
     0.000, 0.000, 0.000,       # att
     0.00001, 0.00001, 0.00001,       # vel
     0.0001, 0.0001, 0.0001,       # accel bias
     0.0001, 0.0001, 0.0001,    # gyro bias
     0.000001,                    # baro bias
     1.0]                       # reference altitude

# Constant Parameters
p_b2g: [0, 0, 0] # position from body frame to gps frame

# Initial State
manual_ref_lla: false # Use manual ref_lla or use first lla from gps
ref_lla: [40.267320, -111.635629, 1387.0]
ref_heading: 0.0 # reference heading for inertial frame (from north)

x0: [0, 0, 0,       # pos
     1, 0, 0, 0]    # att

## Sensor Noise
accel_noise_stdev: 0.5 #used in propagation model noise
gyro_noise_stdev: 0.02 #used in propagation model noise
position_noise_stdev: 0.01 # Mocap # used in sensor model noise
attitude_noise_stdev: 0.01 # Mocap # used in sensor model noise
baro_pressure_noise_stdev: 2.0 # used in sensor model noise
range_noise_stdev: 0.1 # used in sensor model noise

manual_compassing_noise: true
rtk_compassing_noise_stdev: 0.1 #randomly selected for now

# used in sensor model noise if set to true
manual_gps_noise: true # choose noise params for GPS or use the values reported from the message topic
gps_horizontal_stdev: 0.021
gps_vertical_stdev: 0.04
gps_speed_stdev: 0.02
